### ISSUE TITLE
is it abandoned? Is master broken?

### ISSUE THREAD

**Issue by fbcosentino** [QuadCubedStudios/GoDAW#61]
Hi! I cloned master as of today (2023-12-28) and tried opening with versions 3.2, 3.3.3 and 3.5.2. None of them work, all give the same error:

```
E 0:00:05.615   emit_signal: Error calling method from signal 'track_pressed': 'WindowDialog::_popup': Method not found..
  <C++ Source>  core/object.cpp:1260 @ emit_signal()
```

Anything involving opening a popup doesn't work.

Did I do something awfully wrong?

**Comment by Technohacker** [comment]
Hey! Yeah we haven't been able to get much time to devote to the project, so progress is unlikely for a while :sweat_smile: 

As for that error, it's probably on our side. Let me check what's going on :thinking:

**Comment by Technohacker** [comment]
Judging by this,
https://github.com/QuadCubedStudios/GoDAW/blob/fa79a42bd5d96f6c7c45d90adb6bf8784b6b484e/Main.tscn#L188
I'm going to hazard a guess and say `_popup` doesn't exist for some reason. Can you try reconnecting that signal?

**Comment by fbcosentino** [comment]
I tried that, and then I get:
```
E 0:00:02.713   emit_signal: Error calling method from signal 'track_pressed': 'WindowDialog::popup': Cannot convert argument 1 from String to Rect2..
  <C++ Source>  core/object.cpp:1242 @ emit_signal()
```

so I guess there is some fixing to do to make it run. If that's the case, wouldn't it be best to spend the effort porting to Godot 4?

I could give it a shot, but since I can't run on Godot 3, I don't know what is the correct behaviour for things so I don't know what to fix.
You don't happen to have a binary release anywhere, do you?

**Comment by Technohacker** [comment]
Agreed, if/when we get some time I'm open to implementing it in 4.x instead :)

Tbf there's not a whole lot actually implemented, at most there's some code for writing instruments and a basic GDScript-based song scripting system. We don't have much logic implemented for drawing songs on a UI from what I remember :sweat_smile:
